docs/docs/routing.md

```diff
@@ -0,0 +1,241 @@
+
+Adding support for HTML5 History involves three steps:
+
+1. Exporting a history wrapper object
+2. Ensuring links call the History API instead of reloading the page
+3. Updating your rendered content when the current URL changes
+
+## History wrapper objects
+
+The HTML5 History API is not without issues.
+
+Firstly, there are inconsistencies between how the different browsers emit navigation events. Notably, some versions of Chrome and Safari will emit an event on page load while other versions won't.
+
+Secondly, navigation events are only emitted when they're caused by external stimuli such as the *Back* or *Forward* buttons. When the application itself updates the URL, nothing will happen. This means that the application will need to manually pass the new URL from the link that caused it to the top-level component that is handling routing.
+
+Typically, these issues are solved by creating a wrapper around the HTML5 History API. This wrapper will do two things:
+
+- It will normalise events between browsers
+- It will emit navigation events caused by both the application and external stimuli
+
+While it is certainly possible to create your own wrapper, most React applications in the wild make use of the [history](https://github.com/mjackson/history) package.
+
+```bash
+npm install history --save
+```
+
+As the History API is global, the wrapper will also need to be accessible everywhere. A good place to put it is in a `historyWrapper.js` file at the root of your source tree.
+
+```js
+import createBrowserHistory from 'history/createBrowserHistory';
+
+const historyWrapper = createBrowserHistory();
+
+export default historyWrapper;
+```
+
+This guide will use the following methods and properties from the wrapper object:
+
+```js
+// Get the current location.
+const location = historyWrapper.location
+
+// Listen for changes to the current location.
+const unlisten = historyWrapper.listen((location) => {
+  // location is an object like window.location
+  console.log(location.pathname)
+})
+
+// Change current location, adding a new entry to the
+// browser history.
+historyWrapper.push('/home')
+```
+
+You can view the documentation for the full wrapper API at the [history](https://github.com/mjackson/history) package's page.
+
+## Links
+
+By default, HTML `<a>` elements will always cause the page to reload when clicked. If you'd like to update the URL using the History API, you'll need to cancel the default behaviour and then implement an alternative.
+
+While you could do this by adding an `onClick` handler to each individual `<a>` element, React applications often use a `Link` component. Instances of this component behave exactly like `<a>` elements, except that instead of refreshing the page, they will update the URL in-place using `historyWrapper.push()`:
```

Comment:
It sounds a bit like `Link` component already exists. I would rephrase to say that since most links in the app share the same logic, people often create a component for them.